src/agoradatatools/process.py

```diff
@@ -16,59 +16,59 @@
+            logger.warning(
+                f"No custom transformation function provided for dataset {dataset_name}. Skipping."
+            )
         return None
-    if dataset_name == "biodomain_info":
-        return transform.transform_biodomain_info(datasets=datasets)
-    if dataset_name == "genes_biodomains":
-        return transform.transform_genes_biodomains(datasets=datasets)
-    if dataset_name == "overall_scores":
-        df = datasets["overall_scores"]
-        return transform.transform_overall_scores(df=df)
-    if dataset_name == "distribution_data":
-        return transform.transform_distribution_data(
-            datasets=datasets,
-            overall_max_score=dataset_obj["custom_transformations"][
-                "overall_max_score"
-            ],
-            genetics_max_score=dataset_obj["custom_transformations"][
-                "genetics_max_score"
-            ],
-            omics_max_score=dataset_obj["custom_transformations"]["omics_max_score"],
-        )
-    if dataset_name == "team_info":
-        return transform.transform_team_info(datasets=datasets)
-    if dataset_name == "rnaseq_differential_expression":
-        return transform.transform_rnaseq_differential_expression(datasets=datasets)
-    if dataset_name == "gene_info":
-        return transform.transform_gene_info(
-            datasets=datasets,
-            adjusted_p_value_threshold=dataset_obj["custom_transformations"][
-                "adjusted_p_value_threshold"
-            ],
-            protein_level_threshold=dataset_obj["custom_transformations"][
-                "protein_level_threshold"
-            ],
-        )
-    if dataset_name == "rna_distribution_data":
-        return transform.transform_rna_distribution_data(datasets=datasets)
-    if dataset_name == "proteomics_distribution_data":
-        return transform.transform_proteomics_distribution_data(datasets=datasets)
-    if dataset_name in ["proteomics", "proteomics_tmt", "proteomics_srm"]:
-        df = datasets[dataset_name]
-        return transform.transform_proteomics(df=df)
-    if dataset_name in ["biomarkers", "pathology"]:
-        return transform.immunohisto_transform(
-            datasets=datasets, dataset_name=dataset_name
-        )
-    if dataset_name == "model_details":
-        return transform.transform_model_details(datasets=datasets)
+    if isinstance(function_info, str):
+        function_name = function_info
+        config_defined_params = {}
     else:
-        return None
+        # Retrieve the function name and its parameters
+        # Assumes a single function in the dict
+        function_name, config_defined_params = next(iter(function_info.items()))
+    try:
+        retrieved_function = getattr(transform, function_name)
+    except AttributeError:
+        raise AttributeError(
+            f"Function {function_name} not found in the transform module. Please provide the correct function name."
+        )
```

Comment:
NIT: I usually try to avoid nesting code whenever possible. I often think about this video that Bryan shared a while back https://www.youtube.com/watch?v=CFRhGnuXG-4.


```suggestion
    if not hasattr(transform, function_name):
        raise AttributeError(
            f"Function {function_name} not found in the transform module. Please provide the correct function name."
        )
    
    retrieved_function = getattr(transform, function_name)
```